pkgs/development/python-modules/lintrunner/default.nix

```diff
@@ -0,0 +1,55 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  maturin,
+  rustPlatform,
+  cargo,
+  rustc,
+  python,
+}:
+
+buildPythonPackage rec {
+  pname = "lintrunner";
+  version = "0.12.7";
+  format = "pyproject";
+
+  # NOTE: there is a PR open on the upstream repository for the Cargo.lock file.
+  # After that is merged we can change the owner and rev to the upstream
+  # repository instead of a fork.
+  src = fetchFromGitHub {
+    owner = "PrestonHager";
+    repo = pname;
+    rev = "main";
+    #rev = "v${version}";
+    hash = "sha256-CYSNEDT9RO7bnVW8zcDDJOva4GOKGMNs1CcU//IbBrc=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-dv/CznOB/VFx1PM+cYmdHuiK7d8z/HUQbDt5iy5LUTI=";
+  };
+
+  # Adding maturninBuildHook will automatically build and install the executable
+  nativeBuildInputs = [
+    maturin
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+    cargo
+    rustc
+  ];
+
```

Comment:
Could you add tests with `pythonImportsCheck` and `pytestCheckHook`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

Reply:
Since it's an application, there's no imports to check with python. It's written in rust, should I run `cargo test` under the package to ensure tests? Not quite the same as testing the installed binary, but testing the application code functionality itself.